R/distance.R

```diff
@@ -0,0 +1,192 @@
+#' @name distance
+#' @title Spectra Distance/Similarity Measurements
+#'
+#' @description
+#' These functions provide different similariy/distance measurements.
+#'
+#' @param x `matrix`, two-columns e.g. m/z, intensity
+#' @param y `matrix`, two-columns e.g. m/z, intensity
+#'
+#' @details
+#'
+#' @return `double(1)` value between `0:1`, where `0` is completely different
+#' and `1` identically.
+#'
+#' @rdname distance
+#' @author Sebastian Gibb
+#' @family distance/similarity functions
+#' @references
+#' Stein, S. E., and Scott, D. R. (1994).
+#' Optimization and testing of mass spectral library search algorithms for
+#' compound identification.
+#' Journal of the American Society for Mass Spectrometry, 5(9), 859--866.
+#' \doi{10.1016/1044-0305(94)87009-8}.
+#'
+#' Horai et al. (2010).
+#' MassBank: a public repository for sharing mass spectral data for life
+#' sciences. Journal of mass spectrometry, 45(7), 703--714.
+#' \doi{10.1002/jms.1777}.
+#'
+#' Toprak et al. (2014).
+#' Conserved peptide fragmentation as a benchmarking tool for mass spectrometers
+#' and a discriminating feature for targeted proteomics.
+#' Molecular & Cellular Proteomics : MCP, 13(8), 2056--2071.
+#' \doi{10.1074/mcp.O113.036475}.
+#'
+#' @examples
+#'
+#' x <- matrix(c(1:5, 1:5), ncol = 2, dimnames = list(c(), c("mz", "intensity")))
+#' y <- matrix(c(1:5, 5:1), ncol = 2, dimnames = list(c(), c("mz", "intensity")))
+NULL
+
+#' @rdname distance
+#'
+#' @param m `numeric`, weighting for the first column of `x` and `y` (e.g.
```

Comment:
Could you explain that parameter a little more? Is this a single numeric representing the weight for the complete first column or weights for each row? How is that used in the calculation?

Reply:
It is specified in the details section of `ndotproduct` (and every other details section has a "please see details section of `ndotproduct`" line in it). Should I add a line "Please see `ndotproduct` details" or explain the arguments here (I am in favor of the "Please see" line)?

Reply:
Yep I would do the same. Please add the *for more detail see*

Reply:
Done